tests/base_test.py

```diff
@@ -0,0 +1,106 @@
+from flask import Flask, jsonify, url_for
+import unittest
+from prometheus_client import CollectorRegistry, Counter, make_wsgi_app
+from flask_monitor import register_metrics, watch_dependencies
+from werkzeug.middleware.dispatcher import DispatcherMiddleware
+
+class FlaskMonitorTestCase(unittest.TestCase):
+    def setUp(self):
+        """ setUp test """
+        self.registry = CollectorRegistry()
+        self.app = Flask(__name__)
+        self.prometheus = make_wsgi_app()
+
+
+        @self.app.route('/')
+        @self.app.route('/teste')
+        def index():
+            """ Home """
+            return jsonify({'home':True})
+
+        def check_db():
+            """ HealthCheck """
+            return True
+        def is_error200(code):
+            """ Check error """
+            code = str(code) if type(code) is int else code
+            return code.startswith("2")
+
+        @self.app.route('/metrics')
+        def metrics():
+            """ Default metrics """
+            return self.dispatcher.mounts['/metrics']({}, self.capture)[0]
+
+        register_metrics(self.app, registry=self.registry, error_fn=is_error200)
+        self.dispatcher = DispatcherMiddleware(self.app.wsgi_app, {"/metrics": make_wsgi_app(registry=self.registry)})
+        # self.app.wsgi_app = DispatcherMiddleware(self.app.wsgi_app, self.registry)
+        self.thread = watch_dependencies("database", check_db, time_execution=1, registry=self.registry, app=self.app)
```

Comment:
@Ziul, could you please add tests to the `watch_dependencies` func as well?

Reply:
Done @abilioesteves !